zest/releaser/utils.py

```diff
@@ -494,6 +518,47 @@ def system(command, input=''):
     return result
 
 
+def retry_command(command):
+    """Run the command and possibly retry it.
+
+    Print interesting lines.
+
+    When we discover errors, we give three options:
+    - Abort
+    - Retry
+    - Continue
+
+    There is an error is there is a red color in the output.
+
+    It might be a warning, but we cannot detect the distinction.
+    """
+    result = system(command)
+    if Fore.RED not in result:
+        show_interesting_lines(result)
+        return result
+    # There are warnings or errors. Print the complete result.
+    print(result)
+    question = ("There were errors or warnings. Are you sure "
+                "you want to continue? [Yes/No/Retry]")
+    if AUTO_RESPONSE:
+        msg = ("The question '%s' requires a manual answer, but "
+               "we're running in --no-input mode.")
+        msg = msg % question
+        raise RuntimeError(msg)
+    while True:
+        input = get_input(question)
+        if input:
+            input = input.lower()
+            if input == 'y':
+                # Accept the error, continue with the program.
+                return result
+            if input == 'n':
+                sys.exit(1)
```

Comment:
The UX that would make most sense to me is:

> Uploading to PyPI failed.  Retry? [Y/n/q/?]

where the choices are:

Yes -- retry the upload (default, so I can just hit Enter)

No -- don't retry, but continue with the process (which is safe, since zest.releaser will ask me about doing stuff like git pushes so if I wanted to abort, I can still hit ^C then) -- presumably I did a twine upload in a different terminal or something; this is the previous behaviour.

Quit - abort everything.  I'm not sure we even need this, ^C would work.

? would explain the single-letter choices.

Also, a more useful abort (one that would justify a separate option here) would be one that undid the previously-made commits with the version number etc and removed any tags that were created (in case tags get created prior to the upload; I don't remember).  It's something that sounds like a bit of work that can maybe wait for a separate PR.